cmd/zed/agents/zfs_diagnosis.c

```diff
@@ -813,12 +821,32 @@ zfs_fm_recv(fmd_hdl_t *hdl, fmd_event_t *ep, nvlist_t *nvl, const char *class)
 			}
 
 			if (zcp->zc_data.zc_serd_checksum[0] == '\0') {
+				if (nvlist_lookup_uint64(nvl,
+				    FM_EREPORT_PAYLOAD_ZFS_VDEV_CKSUM_N,
+				    &checksum_n) != 0) {
+					checksum_n = UINT64_MAX;
+				}
+				if (nvlist_lookup_uint64(nvl,
+				    FM_EREPORT_PAYLOAD_ZFS_VDEV_CKSUM_T,
+				    &checksum_t) != 0) {
+					checksum_t = UINT64_MAX;
+				}
+				/*
+				 * UINT64_MAX is the default value of the
+				 * checksum_* vdev properties to represent a
+				 * 'none' value.
+				 */
+				if (checksum_n == UINT64_MAX)
+					checksum_n = DEFAULT_CHECKSUM_N;
+				if (checksum_t == UINT64_MAX)
+					checksum_t = DEFAULT_CHECKSUM_T;
+
```

Comment:
This is a bit awkward. Would it be better to leave these fields out of the payload if they are unset?

Reply:
yea, that is a bit weird....maybe better to leave out when unset, that way the semantics of these internal values don't bleed into userland.

it looks like `sys/fs/zfs.h` is a common header between ZED, `module/zcommon/zpool_prop.c`, `module/zfs/vdev.c`, and `module/zfs/zfs_fm.c`...although its not explicitly declared in `zfs_fm.c` but brought in through other includes.

..eh, probably better to just drop these fields from the payload when unset and let ZED define the default - I'll update this PR to reflect your suggestion